core-parser/src/test/java/org/uniprot/core/parser/fasta/uniprot/UniProtKBFastaParserTest.java

```diff
@@ -221,6 +222,24 @@ void canParseToFastaWithMultipleOrfGeneNames() {
         assertTrue(fastaValue.contains("AAAAAAAAAABBBBBBBBBBAAAAAAAAAA"));
     }
 
+    @Test
+    void testToUniProtKBFasta(){
+        List<Gene> genes = List.of(new GeneBuilder().orfNamesAdd(new ORFNameBuilder().value("ORF 1").build()).build());
+        UniProtKBEntry entry = getEntryWithGene(genes);
+        UniProtKBFasta result = UniProtKBFastaParser.toUniProtKBFasta(entry);
+        assertNotNull(result);
+        assertEquals(entry.getPrimaryAccession().getValue(), result.getId());
+        assertEquals(entry.getEntryType(), result.getEntryType());
+        assertEquals(entry.getUniProtkbId(), result.getUniProtkbId());
+        assertEquals(getProteinDescription(entry.getProteinDescription()), result.getProteinName());
```

Comment:
Ideally, you should not depend on the methods of the subject class for assertion purposes, instead just put your expected value if possible.